src/huggingface_hub/hf_api.py

```diff
@@ -2347,6 +2348,7 @@ def file_exists(
             return False
 
     @validate_hf_hub_args
+    @_deprecate_method(version="1.0", message="Use `list_repo_tree` and `get_paths_info` instead.")
```

Comment:
I don't expect this method to be extensively used. It is ok to not wait for `1.0` but rather remove it after a few minor versions.

```suggestion
    @_deprecate_method(version="0.23", message="Use `list_repo_tree` and `get_paths_info` instead.")
```